app/models/schema/common.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+##
+# Defining a common superset of attributes for Resources
+module Schema
+  module Common
+    extend ActiveSupport::Concern
+
+    ##
+    # Retrieve the attributes from Hyrax, imported from external systems, and local
+    def self.attributes
+      Plum.attributes + Schema::IIIF::Book.attributes
+    end
+
+    included do
+      Common.attributes.each do |field|
+        attribute field
```

Comment:
You could always make them not modules - add a method like `apply_schema` to Valhalla::Resource (or via a module I guess) and each schema be a class with an interface that `apply_schema` can utilize.